### DISCUSSION TITLE
clean up after build

### DISCUSSION THREAD

**Discussion by hmoffatt** [buildbot/buildbot#6826] [Q&A]
Is there an easy work to clean up after a build finishes, ie removing all generated files? Other than repeating the source checkout with the appropriate clobber mode set.

**Comment by akshayamadhuri** [comment]
Yes, there are a few ways to clean up after a build finishes and remove all generated files.

Using build systems: Most build systems like Make, CMake, and Gradle provide a way to clean up after a build. .

Using shell scripts: You can create a shell script that removes all generated files after a build finishes. 

Using version control systems: If you're using a version control system like Git, you can simply revert the workspace to the last commit to remove all generated files.
